osu.Game.Rulesets.Taiko/Objects/Drawables/DrawableDrumRoll.cs

```diff
@@ -95,15 +103,27 @@ protected override DrawableHitObject CreateNestedHitObject(HitObject hitObject)
         protected override SkinnableDrawable CreateMainPiece() => new SkinnableDrawable(new TaikoSkinComponent(TaikoSkinComponents.DrumRollBody),
             _ => new ElongatedCirclePiece());
 
-        public override bool OnPressed(TaikoAction action) => false;
+        public override bool OnPressed(TaikoAction action)
+        {
+            if (judgingStarted)
+                OnHit.Invoke(action, HitObject.IsStrong);
```

Comment:
I still would like to see this removed. All changes to this file are unnecessary and you can use the `OnNewResult` method already inside `TaikoPlayfield` instead.

Reply:
As an aside, this _does_ actually need to be changed, because as it is this is providing extra feedback where hits don't exist: https://drive.google.com/file/d/1jII1vYyuDi182Sx0ThRVQnlMF4dnFu55/view?usp=sharing

Reply:
Hang on. We might have different ideas of what's supposed to be happening here. 😅

I've been aiming to make it that a note spawns and plays whenever the user hits the drum inside of a drumroll, regardless of scoring or judging. If I move the logic to `OnNewResult`, it will only notes when a drumroll tick object has successfully judged. On drumrolls where the ticks are fairly spaced out (Like in the test scene), having notes only play periodically would feel broken (But maybe in more normal cases, the ticks will always be close enough).

Is that how osu!taiko classic worked? I've tried looking at a lot of the gameplay videos, but I can't tell. In OG Taiko, the notes always play per hit inside a drumroll (But admittedly the scoring is completely different).

Reply:
That's how osu! works. You have to play the drumroll in time to the music, not just smash keys.

Reply:
Ahhh right! Okay! My mistake! Sorry about that! I should have properly clarified that.
Okay then. Lemme clean out the drumroll object and move all the logic up to the playfield then.